core/nightshadev2/src/nightshade.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::HashSet;
+use std::cmp::Ordering;
+
+pub type AuthorityId = usize;
+pub type BLSSignature = u64;
+
+pub enum NSResult{
+    Updated(Option<State>),
+    Error(String),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct BareState {
```

Comment:
This needs comments what is `BareState` vs `State`?

Reply:
I added some comments, though it is a good idea to add an overview of the consensus at the beginning of the file.

Reply:
Move field-specific comments to each field, instead of having them all at the top of the struct. There is certain format of the documentation that most of Rust-friendly tools expect. It is very likely that we will eventually be using API documentation generation tools that expect this format, so it is a good idea to annotate things now. You can use quick help in CLion to interactive display documentation for the highlighted objects. The rule of thumb that I use myself is that when I work with the code an I highlight some object and expect it to be explained (just like std objects) but there is no interactive help for it then I go and add help. See the network crate for an example of how things should be commented. Avoid tautology, though.

Reply:
Some comments [were added](https://github.com/nearprotocol/nearcore/pull/549/commits/71da4a93bbe3c50779b5bade05086343db457644)